staging/src/k8s.io/apimachinery/pkg/apis/config/types.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
```

Comment:
Is this the best possible name for this package?  Should it be client or clientconfig or something?  "config" is REALLY nondescript.

Reply:
The package is generic and may hold any generic componentconfig type. I think config is the best we can do here. We may always rename later before going beta with the package structure